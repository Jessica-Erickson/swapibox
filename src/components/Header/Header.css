@font-face {
  font-family: 'StarJedi', sans-serif;
  src: url(../../assets/fonts/Starjedi.ttf) format('truetype');
}

.Header {
  grid-row: 1 / 2;
  grid-column: 2 / 3;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.upper-header {
  width: 90%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header-title {
  font-family: StarJedi, sans-serif;
  color: gold;
}

nav {
  width: 70%;
  display: flex;
  justify-content: space-around;
  align-items: center;
}