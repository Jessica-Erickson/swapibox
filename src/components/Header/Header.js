import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({ display }) => {
  return (
    <header className={ display ? 'Header' : 'display-none'}>
      <div className='upper-header'>
        <div></div>
        <NavLink to='/' className='header-title'>
          <h1>SWApi-Box</h1>
        </NavLink>
        <NavLink to='/favorites' className='Button favorites'>
          Favorites
        </NavLink>
      </div>
      <nav>
        <NavLink to='/people' className='Button people'>
          People
        </NavLink>
        <NavLink to='/planets' className='Button planets'>
          Planets
        </NavLink>
        <NavLink to='/vehicles' className='Button vehicles'>
          Vehicles
        </NavLink>
      </nav>
    </header>
  )
}

Header.propTypes = {
  display: PropTypes.bool.isRequired
}

export default Header;