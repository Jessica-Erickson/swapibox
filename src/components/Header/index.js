import React from 'react'
import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import Button from '../Button'
import './Header.css'

const Header = ({ favorites, handleNavClick, currentDisplay }) => {
  return (
    <header className="Header">
      <div className="upper-header">
        <div></div>
        <NavLink to='/' className="header-title">
          <h1>SWApi-Box</h1>
        </NavLink>
        <Button
          label='Favorites'
          favorites={favorites}
          handleNavClick={handleNavClick}
          isActive={'favorites' === currentDisplay}
        />
      </div>
      <nav>
        <Button
          label='People'
          handleNavClick={handleNavClick}
          isActive={'people' === currentDisplay}
        />
        <Button
          label='Planets'
          handleNavClick={handleNavClick}
          isActive={'planets' === currentDisplay}
        />
        <Button
          label='Vehicles'
          handleNavClick={handleNavClick}
          isActive={'vehicles' === currentDisplay}
        />
      </nav>
    </header>
  )
}

Header.propTypes = {
  favorites: PropTypes.number.isRequired,
  handleNavClick: PropTypes.func.isRequired,
  currentDisplay: PropTypes.string.isRequired
}