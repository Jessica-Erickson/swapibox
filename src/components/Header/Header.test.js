import React from 'react'
import Header from './index'
import { shallow } from 'enzyme'

describe('Header', () => {
  let wrapper;
  let favorites;

  beforeEach(() => {
    favorites = []
    wrapper = shallow(<Header 
                        favorites={favorites}
                        handleNavClick={() => {}} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})