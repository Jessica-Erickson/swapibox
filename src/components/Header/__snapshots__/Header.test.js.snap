// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header
  className="Header"
>
  <div
    className="upper-header"
  >
    <div />
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="header-title"
      to="/"
    >
      <h1>
        SWApi-Box
      </h1>
    </NavLink>
    <Button
      favorites={0}
      handleNavClick={[MockFunction]}
      isActive={false}
      label="Favorites"
    />
  </div>
  <nav>
    <Button
      handleNavClick={[MockFunction]}
      isActive={false}
      label="People"
    />
    <Button
      handleNavClick={[MockFunction]}
      isActive={false}
      label="Planets"
    />
    <Button
      handleNavClick={[MockFunction]}
      isActive={false}
      label="Vehicles"
    />
  </nav>
</header>
`;
